controller/workitem.go

```diff
@@ -110,6 +111,37 @@ func (c *WorkitemController) List(ctx *app.ListWorkitemContext) error {
 		exp = criteria.And(exp, criteria.Equals(criteria.Field(workitem.SystemState), criteria.Literal(string(*ctx.FilterWorkitemstate))))
 		additionalQuery = append(additionalQuery, "filter[workitemstate]="+*ctx.FilterWorkitemstate)
 	}
+	if ctx.FilterCategory != nil {
+		var relationships []*category.WorkItemTypeCategoryRelationship
+		err := application.Transactional(c.db, func(tx application.Application) error {
+			// Load all workitemtypes related to the specific category
+			relationships, err = tx.Categories().LoadAllRelationshipsOfCategory(ctx, *ctx.FilterCategory)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"category_id": *ctx.FilterCategory,
+					"err":         err,
+				}, "failed to list work items in category")
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
+			}
+			return nil
+		})
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("failed to list work items in category %v", *ctx.FilterCategory)))
```

Comment:
@hectorj2f 
> Wrap the error just one time, you are doing it twice

I disagree with that. One should wrap errors as many times as it makes sense because it lets you annotate the error. If you only annotate the error in the deepest function, then all the different call paths to that function have only one annotation. For example, it makes sense to annotate a file not found error, then annotate that error in the caller as something like "unable to copy", in the caller of that function you might want to annotate it again with "unable to backup your data".